### PR TITLE
blog/content/post/2024-07-25-odroid-h4-getting-started.md: Add blog post

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

**Comment by miczyg1** [comment]
I still need to catch whole setup with RTE and with pomona. I haven't yet tried to access flash with RTE.

File: blog/content/post/2024-07-25-odroid-h4-getting-started.md [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+H4+](https://www.hardkernel.com/shop/odroid-h4-plus/). It is a nice, compact
+and power saving board with Alder Lake-N SoC. But that is not what makes it
+really outrageous. In the hardware and firmware world, having a public
+schematics is a rarity, almost non-existent, but this board actually [have
+one](https://wiki.odroid.com/_media/odroid-h4/hardware/adln-h4_sch_2024-0306.pdf)!
+Yes! Imagine what a firmware developer thinks seeing this... Of course,
+nothing else than open-source firmware port, obviously. In this blog post I
+would like to show you the perspective of firmware porting from scratch,
+beginning with the starting steps and the necessary preparation work you
+should do each time when dealing with new hardware.
```
Comment (pietrushnic) [inline blog/content/post/2024-07-25-odroid-h4-getting-started.md:36]: @miczyg1 maybe linking [this forum thread](https://forum.odroid.com/viewtopic.php?f=171&t=48691) for better understanding of our plans regarding OST2 and Dasharo would be good?

Also pointing to Dasharo Roadmap

File: blog/content/post/2024-07-25-odroid-h4-getting-started.md [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+damage, before you start being rough with it. And believe me, we will (just a
+bit). Let's plug one of the coin cell batteries so that our BIOS setting will
+not get reset every boot (typically a CMOS battery reset will cause booting
+with default settings, like in last century PCs). Also connect the eMMC module
+to check if it is detected properly by the BIOS and OS.
+
+![Plugging eMMC](/img/odroid/odroid_emmc_plug.jpg)
+
+First boot may take a couple of minutes and include board resets, so be
+patient. The board reste may be observed by the blue LEDs going off. When they
```
Comment (pietrushnic) [inline blog/content/post/2024-07-25-odroid-h4-getting-started.md:96]: reste?

File: blog/content/post/2024-07-25-odroid-h4-getting-started.md [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+to the flash with working firmware in case something goes wrong. It will help
+a lot. Let's try to switch the flash and boot it to see if there is a valid
+firmware in the second flash chip. So let's swap the jumper near the DC jack
+as the Hardkernel page says:
+
+```txt
+Dual BIOS: If the BIOS is corrupted due to a power outage during update,
+etc., you can boot into the backup BIOS and recover by moving the jumper next
+to the DC jack. This feature is only available on ODROID-H4+ and ODROID
+H4-Ultra.
```
Comment (pietrushnic) [inline blog/content/post/2024-07-25-odroid-h4-getting-started.md:117]: > I'm not sure if quoting would not be better

File: blog/content/post/2024-07-25-odroid-h4-getting-started.md [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+mediums were connected):
+
+![ODROID BIOS setup](/img/odroid/odroid_setup.jpg)
+
+Apparently this board has two BIOS SPI flashes switchable by a jumper. That is
+great! We don't have to worry that much about recovery, we can always switch
+to the flash with working firmware in case something goes wrong. It will help
+a lot. Let's try to switch the flash and boot it to see if there is a valid
+firmware in the second flash chip. So let's swap the jumper near the DC jack
+as the Hardkernel page says:
```
Comment (pietrushnic) [inline blog/content/post/2024-07-25-odroid-h4-getting-started.md:111]: In the theory of operation (probably better for the Lab Assembly Guide or another document with procedures), we should describe how to use it during the development, debugging, and testing of firmware mods.

**Review by pietrushnic — CHANGES_REQUESTED** [review]

**Comment by miczyg1** [comment]
> @miczyg1 I guess dual BIOS does not help here in this case. Most likely during training I would use dual BIOS approach, so it would be great if there would be proof of that approach for porting needs.

Definitely it is convenient for end users which would like to experiment with open-source firmware without bricking their devices. but for development rather not much useful.

**Comment by pietrushnic** [comment]
@miczyg1 why? I understand it is not helpful for professional development when we put things in the lab, but why it cannot be leveraged during training? Setting up Odroid+RTE during training is PITA, but having a jumper (button?) recovery sounds useful, but maybe I'm missing something.

**Comment by miczyg1** [comment]
> @miczyg1 why? I understand it is not helpful for professional development when we put things in the lab, but why it cannot be leveraged during training? Setting up Odroid+RTE during training is PITA, but having a jumper (button?) recovery sounds useful, but maybe I'm missing something.

Count trainees to "end users". Of course it will also be useful for training

File: blog/content/post/2024-07-25-odroid-h4-getting-started.md [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+![Unboxing](/img/odroid/odroid_batteries.jpg)
+
+The board itself is so small that it fits into my hand.
+
+![Unboxing](/img/odroid/odroid_board2.jpg)
+
+And mine already has a RAM stick plugged in. According to board
+specifications, it has a single channel DDR5 memory. In single channel
+configuration it may reach a pretty frequency of up to 4800MHz (depending on
+the RAM stick as well of course).
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply blog/content/post/2024-07-25-odroid-h4-getting-started.md]: https://github.com/3mdeb/news-and-ideas/pull/599/commits/6a76815d92d3056aa5a55ff8fe3c039d8ec526b8#diff-a448115412de00a5d6d849f9b70b6f4a3bd46159e359ff67697ff44f3be6d186R81-R83

File: blog/content/post/2024-07-25-odroid-h4-getting-started.md (hunk 2 above)
Comment (miczyg1) [reply blog/content/post/2024-07-25-odroid-h4-getting-started.md]: https://github.com/3mdeb/news-and-ideas/pull/599/commits/6a76815d92d3056aa5a55ff8fe3c039d8ec526b8#diff-a448115412de00a5d6d849f9b70b6f4a3bd46159e359ff67697ff44f3be6d186R100

File: blog/content/post/2024-07-25-odroid-h4-getting-started.md (hunk 3 above)
Comment (miczyg1) [reply blog/content/post/2024-07-25-odroid-h4-getting-started.md]: https://github.com/3mdeb/news-and-ideas/pull/599/commits/6a76815d92d3056aa5a55ff8fe3c039d8ec526b8#diff-a448115412de00a5d6d849f9b70b6f4a3bd46159e359ff67697ff44f3be6d186R117-R120

File: blog/content/post/2024-07-25-odroid-h4-getting-started.md [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+
+If everything was connected correctly, flashrom should detect a chip. Some
+platform may be tricky and require various conditions for the flash to be
+detected: platform powered on, platform powered off, platform cut off from
+power, platform held in reset state, etc. Try various scenarios and see which
+work for you. If you got the flash chip detected, try to read it a couple
+times and check shasums as before to ensure reliability of the programmer. If
+all shasums are the same, try to write the dump made earlier to confirm the
+flash write also works. Boot the platform after you flash the platform with an
+external programmer to confirm the programmer works reliably. IN this case you
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply blog/content/post/2024-07-25-odroid-h4-getting-started.md]: https://github.com/3mdeb/news-and-ideas/pull/599/commits/6a76815d92d3056aa5a55ff8fe3c039d8ec526b8#diff-a448115412de00a5d6d849f9b70b6f4a3bd46159e359ff67697ff44f3be6d186R290

**Comment by miczyg1** [comment]
Applied Grammarly: 6a76815d92d3056aa5a55ff8fe3c039d8ec526b8

File: blog/content/post/2024-07-25-odroid-h4-getting-started.md (hunk 1 above)
Comment (miczyg1) [reply blog/content/post/2024-07-25-odroid-h4-getting-started.md]: 2e6545c

**Review by pietrushnic — APPROVED** [review]

**Pull request merged**